app/models/manageiq/providers/autosde/storage_manager/host_initiator.rb

```diff
@@ -0,0 +1,21 @@
+class ManageIQ::Providers::Autosde::StorageManager::HostInitiator < ::HostInitiator
+  supports :create
+
+  def self.raw_create_host_initiator(_ext_management_system, _options = {})
+    host_initiator_to_create = _ext_management_system.autosde_client.StorageHostCreate(
+      :name           => _options['name'],
+      :port_type      => _options['port_type'],
+      :storage_system => PhysicalStorage.find(_options['physical_storage_id']).ems_ref,
+      :iqn            => _options['iqn'] || "",
```

Comment:
Its a little strange that these have to be empty strings not `nil` if they aren't present in the options.

Are these values actually required by the API?

Reply:
@agrare these values are required (this was decided by the designer). I actually tried to pass nil at first but it didnt work (why ????), so I had to use empty string.

Reply:
Yeah that's what I figured, I don't know that `""` is a good workaround if the API expects this to be present.  We might want ot make all of these properties required if the API requires it

Reply:
chap_name and secret are optional fields and only relevant for ISCSI and not for FC so I dont think we should set them as required. I know that autosde does required them but I think this is something we should reconsider. @gregoryb-at-ibm what do you think?

Reply:
> chap_name and secret are optional fields and only relevant for ISCSI and not for FC

But does the API fail for FC if chap_name or secret are `nil` ?

Reply:
yes, it must get an empty string

Reply:
That's weird then, if it isn't required for the type it shouldn't fail if it isn't present IMO but if this is what we have to do to work around the API that's okay

Reply:
I agree with you, and I'm going to challenge our current autosde design again (i tried a few weeks ago but maybe I wasnt clear enough). For now the work around is required, but if we change autosde api in the future I will do the required modifications in miq.

Reply:
:+1: